src/lib/condition.js

```diff
@@ -234,11 +281,13 @@ class Condition {
    * @return {String} Condition as a URI
    */
   serializeUri () {
-    return 'cc' +
-      ':' + this.getTypeId().toString(16) +
-      ':' + this.getBitmask().toString(16) +
-      ':' + base64url.encode(this.getHash()) +
-      ':' + this.getMaxFulfillmentLength()
+    const ConditionClass = TypeRegistry.findByTypeId(this.type).Class
+    const includeSubtypes = ConditionClass.TYPE_CATEGORY === 'compound'
+    return 'ni:sha-256;' +
+      base64url.encode(this.getHash()) +
+      '?fpt=' + this.getTypeName() +
+      '&cost=' + this.getCost() +
+      (includeSubtypes ? '&subtypes=' + Array.from(this.getSubtypes()).sort().join(',') : '')
```

Comment:
That's not documented in the RFC at the moment. I'd say we should accept these in any order as long as it's the same set of value.